junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -332,12 +380,16 @@ private void resetPermissionsAndTryToDeleteAgain(Path path, IOException exceptio
 		}
 
 		@SuppressWarnings("ResultOfMethodCallIgnored")
-		private static void resetPermissions(Path path) {
-			File file = path.toFile();
-			file.setReadable(true);
-			file.setWritable(true);
-			if (Files.isDirectory(path)) {
-				file.setExecutable(true);
+		private static void tryToResetPermissions(Path path) {
+			try {
+				File file = path.toFile();
+				file.setReadable(true);
+				file.setWritable(true);
+				if (Files.isDirectory(path)) {
+					file.setExecutable(true);
+				}
+			}
+			catch (UnsupportedOperationException ignore) {
 			}
```

Comment:
I think the `try-catch` block should only be around `path.toFile()` and we should add a comment to the catch block explaining the well-known cases when this happens.

Reply:
Done in 24b6a0b.